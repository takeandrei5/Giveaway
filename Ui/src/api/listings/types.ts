import { PaginatedResult } from '../../utils/types';

export type ItemData = { id: string; image: string; title: string; createdAt: Date };

export type CreateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export type UpdateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export interface FetchListingsResponse {
	listings: PaginatedResult<ItemData>;
}

export type ListingInformation = {
	id: string;
	title: string;
	description: string;
	category: number;
	createdAt: Date;
	images: string[];
};

export type OwnerInformation = {
	email: string;
	name: string;
	image: string;
};

export type FetchListingDetailsResponse = { listingInfo: ListingInformation } & { ownerInfo: OwnerInformation };
