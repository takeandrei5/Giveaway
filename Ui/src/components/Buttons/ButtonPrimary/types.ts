import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = Omit<ButtonProps, 'color'>;
