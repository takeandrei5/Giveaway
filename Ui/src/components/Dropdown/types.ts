export type DropdownProps = {
	id: string;
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
	name: string;
	value: string | number | readonly string[];
	isInvalid?: false;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
