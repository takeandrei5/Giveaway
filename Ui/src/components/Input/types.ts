export type InputProps = {
	id: string;
	name: string;
	placeholder: string;
	value?: string;
	onBlur?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	disabled?: boolean;
	multiline?: boolean;
	isInvalid?: boolean;
	height?: string | number;
	label?: string;
	leftIcon?: JSX.Element;
	rightIcon?: JSX.Element;
	rows?: number;
	width?: string | number;
};
