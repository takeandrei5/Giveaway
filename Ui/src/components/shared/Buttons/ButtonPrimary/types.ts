import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = ButtonProps;
