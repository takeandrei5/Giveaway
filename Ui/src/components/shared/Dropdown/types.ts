export type DropdownProps = {
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
