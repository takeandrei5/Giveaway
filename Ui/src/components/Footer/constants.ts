import { StackItem } from './types';

const stackItems: StackItem[] = [
	{
		stackColumnName: 'Product',
		stackColumnItems: ['Overview', 'Features', 'Tutorials', 'Pricing', 'Releases'],
	},
	{
		stackColumnName: 'Company',
		stackColumnItems: ['About', 'Press', 'Careers', 'Contact', 'Partners'],
	},
	{
		stackColumnName: 'Support',
		stackColumnItems: ['Help Center', 'Terms of Service', 'Legal', 'Privacy Policy', 'Status'],
	},
	{
		stackColumnName: 'Follow us',
		stackColumnItems: ['Facebook', 'Twitter', 'Github', 'Instagram', 'Linkedin'],
	},
];

export { stackItems };
