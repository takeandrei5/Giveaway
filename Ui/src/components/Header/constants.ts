const DEFAULT_AVATAR: string = 'https://avatars.dicebear.com/api/male/username.svg';

export { DEFAULT_AVATAR };
