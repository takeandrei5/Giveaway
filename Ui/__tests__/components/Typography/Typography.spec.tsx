import { Typography } from '@components';
import { BUTTON, CAPTION, DEFAULT, H1, H2, H3, H5, INPUT, PARAGRAPH, SMALL } from '@components/Typography/constants';
import { TypographyProps } from '@components/Typography/types';
import { render } from '@testing-library/react';

describe('Typography', () => {
	it('should match snapshot', () => {
		// Arrange
		const props: TypographyProps = {
			children: 'Test-Text',
			variant: 'input',
		};

		// Act
		const { container } = render(<Typography {...props} />);

		// Assert
		expect(container).toMatchSnapshot();
	});

	describe('variant prop', () => {
		it('should render with variant `h1` styles if variant is h1', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'h1',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', H1.fontSize);
			expect(styles).toHaveProperty('font-weight', H1.fontWeight);
			expect(styles).toHaveProperty('line-height', H1.lineHeight);
		});

		it('should render with variant `h2` styles if variant is h2', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'h2',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', H2.fontSize);
			expect(styles).toHaveProperty('font-weight', H2.fontWeight);
			expect(styles).toHaveProperty('line-height', H2.lineHeight);
		});

		it('should render with variant `h3` styles if variant is h3', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'h3',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', H3.fontSize);
			expect(styles).toHaveProperty('font-weight', H3.fontWeight);
			expect(styles).toHaveProperty('line-height', H3.lineHeight);
		});

		it('should render with variant `h5` styles if variant is h5', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'h5',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', H5.fontSize);
			expect(styles).toHaveProperty('font-weight', H5.fontWeight);
			expect(styles).toHaveProperty('line-height', H5.lineHeight);
		});

		it('should render with variant `button` styles if variant is button', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'button',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', BUTTON.fontSize);
			expect(styles).toHaveProperty('font-weight', BUTTON.fontWeight);
			expect(styles).toHaveProperty('line-height', BUTTON.lineHeight);
		});

		it('should render with variant `input` styles if variant is input', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', INPUT.fontSize);
			expect(styles).toHaveProperty('font-weight', INPUT.fontWeight);
			expect(styles).toHaveProperty('line-height', INPUT.lineHeight);
		});

		it('should render with variant `paragraph` styles if variant is paragraph', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'paragraph',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', PARAGRAPH.fontSize);
			expect(styles).toHaveProperty('font-weight', PARAGRAPH.fontWeight);
			expect(styles).toHaveProperty('line-height', PARAGRAPH.lineHeight);
		});

		it('should render with variant `caption` styles if variant is caption', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'caption',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', CAPTION.fontSize);
			expect(styles).toHaveProperty('font-weight', CAPTION.fontWeight);
			expect(styles).toHaveProperty('line-height', CAPTION.lineHeight);
		});

		it('should render with variant `small` styles if variant is small', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'small',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', SMALL.fontSize);
			expect(styles).toHaveProperty('font-weight', SMALL.fontWeight);
			expect(styles).toHaveProperty('line-height', SMALL.lineHeight);
		});

		it('should render with variant `default` styles if variant is default', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'default',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('font-size', DEFAULT.fontSize);
			expect(styles).toHaveProperty('font-weight', DEFAULT.fontWeight);
			expect(styles).toHaveProperty('line-height', DEFAULT.lineHeight);
		});
	});

	describe('prefix and suffix', () => {
		it('should contain prefix if prefix is not empty or undefined', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
				prefix: 'prefix-test',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			expect(getByTestId('typography').textContent).toBe(`${props.prefix}${props.children}`);
		});

		it('should contain suffix if suffix is not empty or undefined', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
				suffix: 'suffix-test',
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			expect(getByTestId('typography').textContent).toBe(`${props.children}${props.suffix}`);
		});
	});

	describe('center prop', () => {
		it('should render textAlign with value of `center` if center prop is set to true', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
				suffix: 'suffix-test',
				center: true,
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('textAlign', 'center');
		});

		it('should render textAlign with value of `initial` if center prop is set to false', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
				suffix: 'suffix-test',
				center: false,
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('textAlign', 'initial');
		});
	});

	describe('multiline prop', () => {
		it('should render whiteSpace `pre-line` if multiline prop is set to true', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
				suffix: 'suffix-test',
				multiline: true,
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('whiteSpace', 'pre-line');
		});

		it('should render whiteSpace `pre-line` if multiline prop is set to false', () => {
			// Arrange
			const props: TypographyProps = {
				children: 'Test-Text',
				variant: 'input',
				suffix: 'suffix-test',
				multiline: false,
			};

			// Act
			const { getByTestId } = render(<Typography {...props} />);

			// Assert
			const styles: CSSStyleDeclaration = getComputedStyle(getByTestId('typography'));
			expect(styles).toHaveProperty('whiteSpace', 'pre-line');
		});
	});
});
