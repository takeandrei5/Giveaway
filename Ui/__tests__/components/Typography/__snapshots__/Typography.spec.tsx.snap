// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should match snapshot 1`] = `
<div>
  <span
    class="chakra-text css-y4pm85"
    data-testid="typography"
  >
    Test-Text
  </span>
</div>
`;
